### PR TITLE
got test 2 working, test 1 is off by 1, test 3 i have no fucking clue

### PR DESCRIPTION
**Pull request by SammyRobensParadise**

### CODE REVIEW

File: PracticeExercise2/main.cpp [hunk 1]
```diff
@@ -93,18 +91,24 @@ int find_max_sum_of_nodes(BinaryTreeNode *T, int &temp_max_sum) {
     // derive the maximum sum for the right subtree
     int right_sum = find_max_sum_of_nodes(T->right, temp_max_sum);
     // compare T->value, left_sum + T->value, and right_sum + T->value; store as max1
-    int max1;
-    if (left_sum > right_sum) {
-        max1 = left_sum;
-    } else max1 = right_sum;
+    int max1 = T->value;
+    if(left_sum+T->value< max1){
```
Comment (maryrsmcphee) [inline PracticeExercise2/main.cpp]: I think this comparison is backwards, shouldn't max be the greater one?

File: PracticeExercise2/main.cpp [hunk 2]
```diff
@@ -93,18 +91,24 @@ int find_max_sum_of_nodes(BinaryTreeNode *T, int &temp_max_sum) {
...
+        max1 =  left_sum+T->value;
+    }
+    if(right_sum+T->value > max1){
```
Comment (maryrsmcphee) [inline PracticeExercise2/main.cpp:98]: here is better

**Comment by maryrsmcphee** [comment]
For the comparison, I was just about to put up a pr with this (which includes T->value in the comparison)
```
// compare T->value, left_sum + T->value, and right_sum + T->value; store as max1
    int max1;
    if ((left_sum + T->value) > (right_sum + T->value)) {
        max1 = left_sum + T->value;
        if (T->value > max1){
            max1 = T->value;
        }
    } else {
        max1 = right_sum + T->value;
        if (T->value > max1) {
            max1 = T->value;
        }
    }
```

File: PracticeExercise2/main.cpp (hunk 1 above)
Comment (SammyRobensParadise) [reply PracticeExercise2/main.cpp]: yea you are  right, doesnt fix the bug though

File: PracticeExercise2/main.cpp (hunk 2 above)
Comment (SammyRobensParadise) [reply PracticeExercise2/main.cpp:98]: yeah. you are right, doenst fix the bug though

**Review by maryrsmcphee — APPROVED** [review]

**Pull request merged**